### PR TITLE
Fixes for NULL-pointer arguments based on gcc warnings

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Review by medhefgo — APPROVED** [review]
Any extra cflags that were causing these? Cause I'd like to use them. :D

**Comment by evverx** [comment]
@medhefgo I'm planning to add `-finline-limit=1000` to the build test once all warnings like that are fixed. Right now it isn't possible due to stuff like
```sh
In file included from ../src/basic/macro.h:456,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/test/test-dns-domain.c:3:
In function ‘test_dns_label_escape_one’,
    inlined from ‘test_dns_label_escape’ at ../src/test/test-dns-domain.c:180:9:
../src/basic/log.h:209:27: error: ‘%s’ directive argument is null [-Werror=format-overflow=]
  209 |                         ? log_internal(_level, _e, PROJECT_FILE, __LINE__, __func__, __VA_ARGS__) \
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/basic/log.h:231:24: note: in expansion of macro ‘log_full_errno_zerook’
  231 |                 (void) log_full_errno_zerook(level, 0, fmt, ##__VA_ARGS__); \
      |                        ^~~~~~~~~~~~~~~~~~~~~
../src/basic/log.h:238:28: note: in expansion of macro ‘log_full’
  238 | #define log_info(...)      log_full(LOG_INFO,    __VA_ARGS__)
      |                            ^~~~~~~~
../src/test/test-dns-domain.c:168:9: note: in expansion of macro ‘log_info’
  168 |         log_info("%s, %zu, %s, →%d", what, l, expect, ret);
      |         ^~~~~~~~
../src/test/test-dns-domain.c: In function ‘test_dns_label_escape’:
../src/test/test-dns-domain.c:168:28: note: format string is defined here
  168 |         log_info("%s, %zu, %s, →%d", what, l, expect, ret);
      |                            ^~
In file included from ../src/basic/macro.h:12,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/test/test-dns-domain.c:3:
In function ‘test_dns_name_to_wire_format_one’,
    inlined from ‘test_dns_name_to_wire_format’ at ../src/test/test-dns-domain.c:114:9:
../src/test/test-dns-domain.c:72:20: error: argument 2 null where non-null expected [-Werror=nonnull]
   72 |         assert_se(!memcmp(buffer, expect, r));
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~
...
```
and so on.

**Comment by evverx** [comment]
> There's a bunch of valid errors, and AFAICT, some bogus warnings

@keszybz I think it triggers much more bogus "maybe-uninitialized" warnings than usual but with `-O2` they are ignored explicitly by passing `-Wno-maybe-uninitialized` anyway so in theory they shouldn't prevent `-finline-limit` from being rolled out with `--werror`.

File: src/libsystemd/sd-journal/sd-journal.c [hunk 1]
```diff
@@ -1353,10 +1350,10 @@ static int add_any_file(
                 f->close_fd = false; /* Make sure journal_file_close() doesn't close the caller's fd
                                       * (or our own). The caller or we will do that ourselves. */
                 (void) journal_file_close(f);
-                goto finish;
+                goto error;
         }
 
-        close_fd = false; /* the fd is now owned by the JournalFile object */
+        our_fd = -1; /* the fd is now owned by the JournalFile object */
```
Comment (yuwata) [inline src/libsystemd/sd-journal/sd-journal.c]: `TAKE_FD()`?

**Review by yuwata — APPROVED** [review]
LGTM. One minor suggestion.

File: src/libsystemd/sd-journal/sd-journal.c (hunk 1 above)
Comment (keszybz) [reply src/libsystemd/sd-journal/sd-journal.c]: Updated.

**Comment by keszybz** [comment]
focal-amd64 failure was fixed by 2299b1cae32c1fb8911da0ce26efced68032f4f8, let's merge this.

**Pull request merged**